text-serializer-gson-legacy-impl/src/main/java/net/kyori/adventure/text/serializer/gson/legacyimpl/NBTLegacyHoverEventSerializer.java

```diff
@@ -31,14 +31,17 @@
  * A legacy {@link HoverEvent} serializer.
  *
  * @since 4.3.0
+ * @deprecated For removal, use {@link net.kyori.adventure.text.serializer.json.legacyimpl.NBTLegacyHoverEventSerializer}
  */
-public interface NBTLegacyHoverEventSerializer extends LegacyHoverEventSerializer {
+@Deprecated
+public interface NBTLegacyHoverEventSerializer extends net.kyori.adventure.text.serializer.gson.LegacyHoverEventSerializer {
   /**
    * Gets the legacy {@link HoverEvent} serializer.
    *
    * @return a legacy {@link HoverEvent} serializer
    * @since 4.3.0
    */
+  @Deprecated
```

Comment:
Provide `@deprecated` docs in the javadocs for all elements that are deprecated. This applies throughout.